### ISSUE TITLE
Campaigns - Layout assignment permission issues 

### ISSUE THREAD

**Issue by PeterMis** [xibosignage/xibo#1405]
Issue directly related to https://github.com/xibosignage/xibo/issues/1349

While the scenario in that issue will work after the fix, there are still issues with this process.

Imagine we have 3 layouts, our user has permissions to only 2 of them

Layout A, layout **B** and layout C

our user has permissions to Layout A and C but not to B (that was added by administrator).

If our user tries to add a new layout assignment it will add that layout, let's call it Layout D, order will be:
A **B** D C
If user now tries to remove layout D it will not be removed instead campaign will look like that:
A C **B**  D C

In short there is a problem with order, permissions and changes made by the user to layout assignment if the layout to which they do not have full permissions is between/before other layouts.
If the layout (B in this example) is last in the assignment, then user can manage layouts to which he has permissions to just fine, but if it is between 'his' layouts it will  cause problems.
The order assignment also does not always seem to be correctly kept when user does not see all layouts in the campaign.

**Comment by dasgarner** [comment]
I think I would argue (not very strongly I admit) that because the user has Edit permission to the Campaign they should be able to see, reorder and remove the other users layout - but they should not be able to see any other details regarding the Layout (it should just show itself as "Layout (no permission)".

This would solve the above issue as the UI can then determine whether the user is reordering, adding, removing, etc (i.e. it has all of the information to make that decision).

**Comment by dasgarner** [comment]
@PeterMis to review Friday

**Comment by PeterMis** [comment]
I'd be inclined to agree with the above, we shall check and discuss on Friday!
